### DISCUSSION TITLE
FUNC_SAVE_BEFORE_RESTART - it's not only called before restart

### DISCUSSION THREAD

**Discussion by btsimonh** [arendst/Tasmota#21460] [Ideas]
Hi All, 
I was investigating an issue where FUNC_SAVE_BEFORE_RESTART was used to kill stuff off before a restart - a possibly useful feature if whatever the driver is doing is in a separate thread (e.g. BLE and Webcam), which could interfere with the actual shutdown/restart.

However, it's called by SettingsSaveAll(), which is called in far more places than just before shutdown/restart.
I could not find any other command, or any other way to get a trigger when we are about to go dead.

Maybe we should add one?

**Comment by arendst** [comment]
I'll add a new function called FUNC_ABOUT_TO_RESTART ...

**Reply by btsimonh** [reply]
ok, I'll close my stupid PR, an await the new function, then PR with the webcam responding to that after testing.
thankyou,
s

**Comment by arendst** [comment]
Done.
